pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -99,8 +100,10 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 			}},
 		},
 	}
+	logger = logger.WithValues("spc", spc.Name)
```

Comment:
nit: combine this with line 84 and just bring that down?

Reply:
i prefer setting information as soon as we have it

Reply:
why make two separate function calls though? it's effectively doing the same thing, right? just with more lines of code?

Reply:
I like setting information as soon as we have it. Following that pattern makes it easy not to miss information in the future